### PR TITLE
feature/572_batches_within_transactions_test

### PR DESCRIPTION
**Pull request by frbattid**
* Implements issue #572 
* 100% unit test passed:
```
Tests run: 72, Failures: 0, Errors: 0, Skipped: 0
```
* (unofficial) e2e tests passed:
```
time=2015-12-10T14:56:07.047CET | lvl=INFO | trans=1449755758-460-0000000099 | function=persistAggregation | comp=Cygnus | msg=com.telefonica.iot.cygnus.sinks.OrionTestSink[176] : [test-sink] Persisting data at OrionTestSink. Data (Processing event={Processing headers={recvTimeTs=1449755762856, fiwareService=test_service, fiwareServicePath=test_servicepath, destinations=room1_room}, Processing context element={id=Room1, type=Room}, Processing attribute={name=temperature, type=centigrade, value="0", metadata=[{"name":"resolution","type":"integer","value":"0"}]}},Processing event={Processing headers={recvTimeTs=1449755762967, fiwareService=test_service, fiwareServicePath=test_servicepath, destinations=room1_room}, Processing context element={id=Room1, type=Room}, Processing attribute={name=temperature, type=centigrade, value="1", metadata=[{"name":"resolution","type":"integer","value":".50"}]}},Processing event={Processing headers={recvTimeTs=1449755763010, fiwareService=test_service, fiwareServicePath=test_servicepath, destinations=room1_room}, Processing context element={id=Room1, type=Room}, Processing attribute={name=temperature, type=centigrade, value="2", metadata=[{"name":"resolution","type":"integer","value":"1.00"}]}},
...
,Processing event={Processing headers={recvTimeTs=1449755766993, fiwareService=test_service, fiwareServicePath=test_servicepath, destinations=room1_room}, Processing context element={id=Room1, type=Room}, Processing attribute={name=temperature, type=centigrade, value="98", metadata=[{"name":"resolution","type":"integer","value":"49.00"}]}},Processing event={Processing headers={recvTimeTs=1449755767032, fiwareService=test_service, fiwareServicePath=test_servicepath, destinations=room1_room}, Processing context element={id=Room1, type=Room}, Processing attribute={name=temperature, type=centigrade, value="99", metadata=[{"name":"resolution","type":"integer","value":"49.50"}]}})
time=2015-12-10T14:56:07.050CET | lvl=INFO | trans=1449755758-460-0000000099 | function=process | comp=Cygnus | msg=com.telefonica.iot.cygnus.sinks.OrionSink[295] : Finishing transaction (1449755758-460-0000000000,1449755758-460-0000000001,1449755758-460-0000000002,
...
,1449755758-460-0000000098,1449755758-460-0000000099)
```
* Assignee @fgalan

### CODE REVIEW

File: doc/flume_extensions_catalogue/orion_test_sink.md [hunk 1]
```diff
@@ -67,31 +69,39 @@ Assuming the following Flume event is created from a notified NGSI context data
...
-time=2015-11-27T10:41:27.124CET | lvl=INFO | trans=1448617281-654-0000000000 | function=persistOne | comp=Cygnus | msg=com.telefonica.iot.cygnus.sinks.OrionTestSink[109] : [test-sink] Processing context attribute (name=speed, type=float, value=112.9, metadata=[])
-time=2015-11-27T10:41:27.124CET | lvl=INFO | trans=1448617281-654-0000000000 | function=persistOne | comp=Cygnus | msg=com.telefonica.iot.cygnus.sinks.OrionTestSink[109] : [test-sink] Processing context attribute (name=oil-level, type=float, value=74.6, metadata=[])
-time=2015-11-27T10:41:27.124CET | lvl=INFO | trans=1448617281-654-0000000000 | function=process | comp=Cygnus | msg=com.telefonica.iot.cygnus.sinks.OrionSink[178] : Finishing transaction (1448617281-654-0000000000)
+time=2015-12-10T14:31:49.389CET | lvl=INFO | trans=1429535775-308-0000000000 | function=getEvents | comp=Cygnus | msg=com.telefonica.iot.cygnus.handlers.OrionRestHandler[150] : Starting transaction (1429535775-308-0000000000)
```
Comment (fgalan) [inline doc/flume_extensions_catalogue/orion_test_sink.md]: Any actual change in this long long long line ;)?

File: doc/quick_start_guide.md [hunk 2]
```diff
@@ -117,14 +117,11 @@ $ ./notification.sh http://localhost:5050/notify
...
-2015-03-06T08:54:20.803CET | lvl=INFO | trans=1425628437-99-0000000000 | function=persist | comp=Cygnus | msg=com.telefonica.iot.cygnus.sinks.OrionTestSink[90] : [test-sink] Processing context element (id=Room1, type= Room)
-2015-03-06T08:54:20.803CET | lvl=INFO | trans=1425628437-99-0000000000 | function=persist | comp=Cygnus | msg=com.telefonica.iot.cygnus.sinks.OrionTestSink[109] : [test-sink] Processing context attribute (name=temperature, type=centigrade, value=26.5, metadata=[])
-2015-03-06T08:54:20.804CET | lvl=INFO | trans=1425628437-99-0000000000 | function=process | comp=Cygnus | msg=com.telefonica.iot.cygnus.sinks.OrionSink[187] : Finishing transaction (1425628437-99-0000000000)
+time=2015-12-10T14:31:49.389CET | lvl=INFO | trans=1449754282-690-0000000000 | function=getEvents | comp=Cygnus | msg=com.telefonica.iot.cygnus.handlers.OrionRestHandler[150] : Starting transaction (1449754282-690-0000000000)
```
Comment (fgalan) [inline doc/quick_start_guide.md]: Any actual changes in this part?

File: doc/flume_extensions_catalogue/orion_test_sink.md (hunk 1 above)
Comment (frbattid) [inline doc/flume_extensions_catalogue/orion_test_sink.md]: It has chenged the way the received notification is printed :)

File: doc/quick_start_guide.md (hunk 2 above)
Comment (frbattid) [inline doc/quick_start_guide.md]: It has changed the way the received notification is printed.

File: doc/flume_extensions_catalogue/orion_test_sink.md (hunk 1 above)
Comment (fgalan) [inline doc/flume_extensions_catalogue/orion_test_sink.md]: NTC

File: doc/quick_start_guide.md (hunk 2 above)
Comment (fgalan) [inline doc/quick_start_guide.md]: NTC

**Comment by fgalan** [comment]
LGTM

**Pull request merged**